docs/content/design/communication.md

```diff
@@ -0,0 +1,63 @@
+---
+title: "Communication"
+date: 2018-02-11T15:57:56-05:00
+---
+
+## Communication flow
+
+This is the story of a time long ago. A time of myth and legend when the ancient Gods were petty and cruel and they plagued build with irreproducibility.
+Only one project dared to challenge their power...Athens. Athens possessed a strength the world had never seen. Filling its cache.
+
+### Clean plate
+
+At the beginning, there's theoretical state when a cache of proxy and registry is empty.
+
+When User makes a request at this ancient time, it works as described on the flow below.
+
+- User contacts proxy asking for module M in version v1.0
+- Proxy checks whether or not it has this module in its storage. It has not.
+-So it responds with a redirect to Olympus and schedules its own job to check with Olympus in few units of time, giving Olympus space to fill its cache.
+- User receives redirect to Olympus and asks it for module M in version v1.0
+- Olympus is cache free as well, so it asks the underlying VCS (e.g github.com) for a module.
+- After it receives all the bits, it stores it into its own cache and serves it to the User.
+- User receives module and is happy.
+- Sometimes around this time proxy asks Olympus for module M in version v1.0 as well. Olympus, now aware of this module, serves it so proxy can fill its own cache.
```

Comment:
What do you think about saying "Concurrently" instead of "Sometimes around this time"

Also, can you say "the proxy" instead of "proxy"?